ratis-client/src/main/java/org/apache/ratis/client/impl/OrderedStreamAsync.java

```diff
@@ -103,18 +106,27 @@ public String toString() {
   }
 
   private final DataStreamClientRpc dataStreamClientRpc;
-  private final SlidingWindow.Client<DataStreamWindowRequest, DataStreamReply> slidingWindow;
+
   private final Semaphore requestSemaphore;
   private final TimeDuration requestTimeout;
   private final TimeoutScheduler scheduler = TimeoutScheduler.getInstance();
+  private final ClientId clientId;
+
+  private final ConcurrentMap<ClientInvocationId, SlidingWindow.Client<DataStreamWindowRequest, DataStreamReply>>
+      slidingWindows = new ConcurrentHashMap<>();
```

Comment:
We may move slidingWindow to DataStreamOutputImpl.  Then, we don't have to create this map.

Reply:
Has been modified @szetszwo